homeassistant/components/roborock/coordinator.py

```diff
@@ -125,12 +130,21 @@ def _set_current_map(self) -> None:
 
     async def get_maps(self) -> None:
         """Add a map to the coordinators mapping."""
+        self.maps = await self.roborock_storage.async_load_map_info(self.duid)
         maps = await self.api.get_multi_maps_list()
+        update = False
         if maps and maps.map_info:
             for roborock_map in maps.map_info:
-                self.maps[roborock_map.mapFlag] = RoborockMapInfo(
-                    flag=roborock_map.mapFlag, name=roborock_map.name, rooms={}
-                )
+                # To prevent weirdness - if there is a map without a name
+                # - set its name to its flag
+                if roborock_map.mapFlag not in self.maps:
```

Comment:
Writing my thoughts as I reread my code - so stick with me.

We load the saved map info from the storage.

We then get the list of maps from the api. 

If we haven't seen this map before, we should add it to self._maps.

The api doesn't have room information, that information can only be gotten when we have that current map selected. so we can't get it here. instead, room updates would be done in get_rooms().

I think the point about names is fair though, so I will update the name here if the name has changed.

Reply:
Although on second thought - I don't think entities can handle dynamic changing of attribute name, so i'm not sure if that makes sense to handle. I will log a warning for now and you can give input

Reply:
What is purpose is the local disk storage of metadata / map info? The API is always consulted here anyway first and I don't see it read on its own anywhere.

Reply:
So it is important because the only time we can get some of that information is when we are parsing the maps.

So we get the majority of the map info all of the time anyway, but the rooms and a few other attributes are not available from that API - they are only available from the image. So we need to cache it so that if we cannot get the maps on startup, we still have access to room names for other services + entities.